### PR TITLE
feat(1785): Use groupEventId to get parent builds [4]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Would be nice to use `groupEventId` to get parent builds instead of just the immediate `parentBuild` to determine start logic.

## Objective

This PR:
- uses `groupEventId` to get parentBuilds/finishedBuilds info
    - it gets all events with the same groupEventId, removes downstream builds, filters out builds with the same `jobId` to return the most recent build
- fixes some eslint warnings
- adds `prSource` and `prInfo` data to build create

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/1785

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/builds/index.js [hunk 1]
```diff
@@ -363,6 +363,8 @@ async function createInternalBuild(config) {
     } = config;
     const event = await eventFactory.get(build.eventId);
     const prRef = event.pr.ref ? event.pr.ref : '';
+    const prSource = event.pr.prSource ? event.pr.prSource : '';
+    const prInfo = event.pr.prInfo ? event.pr.prInfo : '';
```
Comment (pritamstyz4ever) [inline plugins/builds/index.js:367]:
```suggestion
    const prSource = event.pr.prSource || '';
    const prInfo = event.pr.prInfo || '';
```

File: plugins/builds/index.js [hunk 2]
```diff
@@ -560,15 +564,55 @@ function parseJobInfo({ joinObj, currentJobName, nextJobName, pipelineId, build
...
+            params: {
+                groupEventId: event.groupEventId
+            }
+        });
+        const builds = await event.getBuilds();
+        let parentBuilds = [].concat(builds);
+
+        await Promise.all(
+            parentEvents.map(async pe => {
+                const eventBuilds = await pe.getBuilds();
```
Comment (jithine) [inline plugins/builds/index.js:590]: can we do a single query here ? something like
```
SELECT * FROM  builds 
     WHERE build.eventId IN (
         SELECT event.id FROM events WHERE event.id = groupEventId
     )
     GROUP BY build.jobId

```
and maybe limit as well ?
else we are looking at multiple db queries equivalent to number of events here, which might not be that many in numbers 🤔  exploring if we can offload all this logic to DB

cc @parthasl

Comment (jithine) [reply plugins/builds/index.js:590]: so idea is to add a method in build model to fetch builds  (latest) for jobs for a given groupEventId

Comment (tkyi) [reply plugins/builds/index.js:590]: thinking about using a query like
```
SELECT * FROM (
    SELECT *, RANK() OVER ( PARTITION BY "jobId" ORDER BY "id" DESC ) AS rank
    FROM public.builds WHERE "eventId" IN 
        (SELECT "id" FROM public.events WHERE "groupEventId" = '12345')) AS foo
WHERE rank = 1
```
This should return an _array of builds_ consisting of the latest build for each job that ran in an event that has a matching `groupEventId` 12345

Comment (tkyi) [reply plugins/builds/index.js:590]: https://github.com/screwdriver-cd/models/pull/450

**Pull request merged**